### ISSUE TITLE
Curved path and Gradient for ChartAreasView

### ISSUE THREAD

**Issue by umair-citrusbits** [i-schuetz/SwiftCharts#202]
How to create a curved path ChartAreasView along with gradient color of ChartAreasView

If this is not possible at least make the display function of ChartPointsLayer class public so we can override it and customize the area chart

**Comment by i-schuetz** [comment]
`ChartAreasView` can generate only straight lines at the moment. There's no functionality for curved areas yet, also no gradients.

To solve this I'd adjust `ChartAreasView` and `ChartPointsAreaLayer` to receive a `ChartLinesViewPathGenerator`, which generates the path for `ChartAreasView`. Which you can then pass from where you use the chart like in `CubicLinesExample`.

For the gradient, I'd just add this functionality to `ChartAreasView`, allow to pass an array of colors instead of only one color and create the gradient accordingly. If it's only one color then there's no gradient.

Basically, it would be great if you could do this directly in the library as this functionality is very needed but I hadn't time to implement it. Happy to assist further.

I anyways made `display` open (it's in master). It should be of course possible to create custom layers.
